src/bmp.imageio/bmpinput.cpp

```diff
@@ -184,10 +188,18 @@ BmpInput::read_native_scanline(int subimage, int miplevel, int y, int /*z*/,
         }
     }
     if (m_dib_header.bpp == 8) {
-        for (unsigned int i = 0, j = 0; j < scanline_bytes; ++i, j += 3) {
-            mscanline[j]     = m_colortable[fscanline[i]].r;
-            mscanline[j + 1] = m_colortable[fscanline[i]].g;
-            mscanline[j + 2] = m_colortable[fscanline[i]].b;
+        if (m_allgray) {
+            // Keep it as 1-channel image because all colors are gray
+            for (unsigned int i = 0; i < scanline_bytes; ++i) {
+                mscanline[i] = m_colortable[fscanline[i]].r;
+            }
+        } else {
+            // Expand palette image into 3-channel RGB (existing code)
+            for (unsigned int i = 0, j = 0; j < scanline_bytes; ++i, j += 3) {
+                mscanline[j]     = m_colortable[fscanline[i]].r;
+                mscanline[j + 1] = m_colortable[fscanline[i]].g;
+                mscanline[j + 2] = m_colortable[fscanline[i]].b;
+            }
         }
     }
     if (m_dib_header.bpp == 4) {
```

Comment:
I don't know if you care (nor if I care), but this only fixes the 8 bits per pixel case. Are you concerned with 4bpp and 1bpp cases to also ensure that if their (even more limited) color map entries are all gray that it should appear to OIIO as if it were a one channel image?

Reply:
No, I do not work with 4bpp or 1bpp images.